src/main/java/reposense/parser/CsvParser.java

```diff
@@ -0,0 +1,126 @@
+
+    /**
+     * Returns a list of {@code RepoConfiguration}, which are the inflated object a line of the csv file.
+     * @throws IOException if user-supplied csv file does not exists or is not readable.
+     */
+    public static List<RepoConfiguration> parse(CliArguments argument) throws IOException {
+        assert (argument != null);
+
+        final Date sinceDate = argument.getSinceDate().orElse(null);
+        final Date untilDate = argument.getUntilDate().orElse(null);
+        final Path path = argument.getConfigFilePath();
+
+        List<RepoConfiguration> repoConfigurations = new ArrayList<RepoConfiguration>();
+        int lineNumber = 1;
+
+        try {
+            // Skips first line, which is the header row
+            final Collection<String> lines = Files.lines(path).skip(SKIP_FIRST_LINE).collect(Collectors.toList());
+
+            for (final String line : lines) {
+                processLine(repoConfigurations, sinceDate, untilDate, line, lineNumber);
+                lineNumber++;
+            }
+
+        } catch (IOException ioe) {
+            throw new IOException(MESSAGE_UNABLE_TO_READ_CSV_FILE);
+        }
+
+        return repoConfigurations;
+    }
+
+    /**
+     * Extracts {@code Author} information from the {@code line}.
+     */
+    private static void processLine(List<RepoConfiguration> repoConfigurations,
+            Date sinceDate, Date untilDate, String line, int lineNumber) {
+        if (!line.isEmpty()) {
+            String[] elements = line.split(Constants.CSV_SPLITTER);
+
+            if (elements.length < GITHUB_ID_POSITION) {
+                // Warns malformed line and skips it
+                System.out.println(String.format(MESSAGE_MALFORMED_LINE_FORMAT, lineNumber, line));
+                return;
+            }
+
+            String organization = elements[ORGANIZATION_POSITION];
+            String repositoryName = elements[REPOSITORY_NAME_POSITION];
+            String branch = elements[BRANCH_POSITION];
+
+            RepoConfiguration config = new RepoConfiguration(organization, repositoryName, branch);
+            int index = repoConfigurations.indexOf(config);
+
+            if (index != -1) {
+                config = repoConfigurations.get(index);
+            } else {
+                repoConfigurations.add(config);
+                config.setSinceDate(sinceDate);
+                config.setUntilDate(untilDate);
+            }
+
+            Author author = new Author(elements[GITHUB_ID_POSITION]);
+            config.getAuthorList().add(author);
+            setDisplayName(elements, config, author);
+            setAlias(elements, config, author);
+        }
+    }
+
+    private static void setDisplayName(String[] elements, RepoConfiguration config, Author author) {
+        // Checks length of the elements array as trailing commas may be omitted for empty columns
+        if (elements.length > DISPLAY_NAME_POSITION && !elements[DISPLAY_NAME_POSITION].isEmpty()) {
```

Comment:
Perhaps use an explanatory variable to reduce the need for comments?
`... = elements.length > DISPLAY_NAME_POSITION && !elements[DISPLAY_NAME_POSITION].isEmpty()`